lib/utils.js

```diff
@@ -252,10 +252,16 @@ exports.files = function (dir, ext, ret) {
     .filter(ignored)
     .forEach(function (path) {
       path = join(dir, path);
-      if (statSync(path).isDirectory()) {
-        exports.files(path, ext, ret);
-      } else if (path.match(re)) {
-        ret.push(path);
+      try {
```

Comment:
Thanks for the comments @boneskull !

watching broken symlinks and their content would be nice, but right now it wouldn't follow the current convention: Watch seems to watch only files which mocha can resolve on startup. Thus i think it would be too much of effort to fix the behaviour only for symlinks. 

I wasn't aware of lstatSync function so i ended up to solution I currently propose in this PR. But indeed lstatSync would be more feasible solution here. I played around with lstatSync a bit and it wouldn't throw at all if the symlink is broken - which is fine in my use case at least. Instead, mocha would fail later on to 'module not found' if it tries to require module from broken symlink, which again sounds like a reasonable behaviour to me. That exception also points out the failing module so it's easy to sort out why it is failing.

So what I propose is that I remove the try-catch and replace statSync with lstatSync here.

Watch option built into mocha is pretty important to us since starting up and setting things up takes pretty long time in larger projects.